src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoCommon.cs

```diff
@@ -25,32 +25,48 @@ internal abstract class JsonPropertyInfoCommon<TClass, TDeclaredProperty, TRunti
             Type parentClassType,
             Type declaredPropertyType,
             Type runtimePropertyType,
-            PropertyInfo propertyInfo,
+            MemberInfo propertyInfo,
```

Comment:
I though about that, but then I need to rename the property to `MemberInfo`. After that the name of `JsonPropertyInfo` looks strange, because it's not for properties only. That's why I did no changes in names.

By the way, the `JsonPropertyNameAttribute` has incorrect name too in that case and should be changed, but it's a public API, and the read-only DOM model has the `JsonProperty` struct.

So I don't know how it should be named and what is appropriate now. Leaving this to @steveharter and @ahsonkhan.

Reply:
> and the read-only DOM model has the JsonProperty struct.

`JsonProperty` is about the key/value pairs in the payload, the fact that a deserialized value would consider it to be a field doesn't really matter.  (ECMA-404 doesn't have a name for the key/value pairs even though RFC 7149 called them "members". Newtonsoft.Json called them JProperty in the graph DOM, and we called them JsonProperty in our flat DOM)

Reply:
If it called a member in RFC, why is it named a property here? Legacy from Newtonsoft? Should then it be renamed to be correct?

/cc @karelz

Reply:
Partially legacy, partially that ECMA-404 has no name for it (me noticing that the RFC had a name for it yesterday was sort of an accident; I usually use ECMA-404 as my source of what JSON is... but I accidentally typed RFC and went with it).

> Should then it be renamed to be correct?

I don't think so.

Of the two competing standard that I've seen (ECMA-404 2nd edition, IETF RFC 8259 (the "Internet Standard" status version of the "Proposed Standard" from IETF RFC 7149) one names it, the other doesn't.  That means that it doesn't have a standard standard name.

So now we pop up to [ECMA-262](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-262.pdf) (ECMAScript / Javascript), where we see

> **4.3.30 property**
>
> part of an object that associates a key (either a String value or a Symbol value) and a value

So the deeper history says that "property" is the correct word (and the RFC is "wrong").  Plus, we're so close to 3.0 releasing that API breaking changes have a very, very high threshold... and this almost certainly wouldn't make it at this point.

Reply:
Okay, let's leave other names as is, but what's about these parameter, member and class? Should they be renamed to `memberInfo`, `MemberInfo`, `JsonMemberInfo`?